kratos/utilities/read_materials_utility.cpp

```diff
@@ -0,0 +1,265 @@
+            if (value.IsBool()) {
+                p_prop->SetValue(variable, value.GetBool());
+            } else if (value.IsInt()) {
+                p_prop->SetValue(variable, static_cast<bool>(value.GetInt()));
+            } else if (value.IsDouble()) {
+                p_prop->SetValue(variable, static_cast<bool>(value.GetDouble()));
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<int> >::Has(variable_name)) {
+            const Variable<int>& variable = KratosComponents<Variable<int>>().Get(variable_name);
+            if (value.IsInt()) {
+                p_prop->SetValue(variable, value.GetInt());
+            } else if (value.IsDouble()) {
+                p_prop->SetValue(variable, static_cast<int>(value.GetDouble()));
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<array_1d<double, 3> > >::Has(variable_name)) {
+            const Variable<array_1d<double, 3>>& variable = KratosComponents<Variable<array_1d<double, 3>>>().Get(variable_name);
+            if (value.IsVector()) {
+                array_1d<double, 3> temp(3, 0.0);
+                const Vector& value_variable = value.GetVector();
+                const std::size_t iter_number = (3 < value_variable.size()) ? 3 : value_variable.size();
+                for (std::size_t index = 0; index < iter_number; index++)
+                    temp[index] = value_variable[index];
+                p_prop->SetValue(variable, temp);
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<array_1d<double, 6> > >::Has(variable_name)) {
+            const Variable<array_1d<double, 6>>& variable = KratosComponents<Variable<array_1d<double, 6>>>().Get(variable_name);
+            if (value.IsVector()) {
+                array_1d<double, 6> temp(6, 0.0);
+                const Vector& value_variable = value.GetVector();
+                const std::size_t iter_number = (6 < value_variable.size()) ? 6 : value_variable.size();
+                for (std::size_t index = 0; index < iter_number; index++)
+                    temp[index] = value_variable[index];
+                p_prop->SetValue(variable, temp);
+            } else {
+                KRATOS_ERROR << "Check you write the value in a correct format: " << value << std::endl;
+            }
+        } else if(KratosComponents<Variable<Vector > >::Has(variable_name)) {
+            const Variable<Vector>& variable = KratosComponents<Variable<Vector>>().Get(variable_name);
+            if (value.IsVector()) {
+                p_prop->SetValue(variable, value.GetVector());
+            } else if (value.IsMatrix()) {
```

Comment:
This should not be necessary, if it is then there is sth wrong with the Parameters

I implemented this function and I am quite sure you should not do this

This can silently change the variable type!

Reply:
Really?, I wanted to be sure

Reply:
Have a look at the tests I wrote:
https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/tests/test_kratos_parameters.py#L473-L582

I struggled a lot with this so I am sure it is working as intended
If sth is a Matrix then IsVector will return false and vise-versa

Also if the user defines a matrix then you are doing some sort of conversion, seems hacky to me